### ISSUE TITLE
RSS feed

### ISSUE THREAD

**Issue by aaronoppenheimer** [Gig-o-Matic/GO3#566]
The old gig-o had an RSS feed for each band, which was the same as the band public feed. Should implement this again.

**Comment by PresidentNick** [comment]
Can the API replace this feature?

**Comment by aaronoppenheimer** [comment]
Well, there are occasionally people interested in putting an RSS feed in a wordpress site. I also think RSS is just a nice thing to support.

In the old gig-o I hand-built an RSS feed, but Django provides a really simple (or at least, is seems simple) set of tools for producing RSS feeds.

**Comment by mmendelson222** [comment]
@aaronoppenheimer yes I would use that.  If there is an RSS feed with the fields separated, perhaps it could be considered the de facto feed for public calendars? Then the whole [issue about calendar feeds](https://github.com/Gig-o-Matic/GO3/issues/648) becomes less important.  I hadn't considered using back links. 

Also I see that all description information for a gig is publicly available - if only through the calendar (and perhaps in the future RSS) feeds. Since you are considering adding fields, would you consider a **band-only description** field that will not be public?  Members of my group have raised security concerns, so I'm looking out for those.  I realize this is getting to be a lot of fields, sorry.

**Comment by aaronoppenheimer** [comment]
Hm. RSS doesn't have all that much extra stuff - name and description, then some things we wouldn't use like author, category, copyright. It doesn't have more granularity than that.

Maybe I'm still not clear about what you need. I'm proposing that there are three feeds:

- A member's calfeed, which has gig information of all the gigs a member wants to see from all their bands
- A band's calfeed, which has only public information for 'confirmed' gigs
- A band's RSS feed, which has the same content as the band's calfeed in RSS format

So a gig would have:

- A title (like it does now): would be included in the band's public iCal, the RSS feed, and the member's calfeed
- A 'details' field (like it does now): would be included only in the member's calfeed and visible on the site to members
- A 'public_description' field (new): would be included in the public calfeed and RSS feed.

Now, an iCal feed event has additional important things - like a date. An RSS item doesn't - it only has a name and a description because it's not really designed as a list of events, it's a list of news articles or similar. So I guess we'll have to put the date into the title or the description. For example, the Title of an RSS item that takes place on August 16 called "Birthday Parade" would be "August 16 - Birthday Parade" or similar. Unless you have other thoughts? The screenshot you pasted in the first post on issue #648 doesn't seem really possible with RSS unless I'm missing something.

A

**Comment by aaronoppenheimer** [comment]
Actually - if there's no public description, it shouldn't default to the details field, since bands put "band only" stuff in there.

**Comment by uneasyalliance** [comment]
Good point.
